### ISSUE TITLE
Shortcuts to stick windows to the right and left side of the screen don't work with Terminal

### ISSUE THREAD

**Issue by commadelimited** [rbreaves/kinto#198]
I'm a brand new Ubuntu user trying to get my machine working like a Mac. One of the shortcuts I use all the time so far is the ability to resize/stick windows to the right and left side of the screen (and fullsize). However since installing Kinto I've noticed that these shortcuts don't work for just the Terminal app. What can I do to add this functionality back?

I'm using Kinto with a default install and I'm using an Apple full-size external keyboard. So for me, the shortcut would be:
* Ctrl + right arrow --> Stick to right side of screen
* Ctrl + left arrow --> Stick to left side of screen
* Ctrl + up arrow --> Full screen
* Ctrl + down arrow --> Restore window to original size and position

Thanks for this software by the way. It's really helping the transition.

**Comment by rbreaves** [comment]
Looking into it that doesn't really seem to be a normal behavior via keyboard shortcuts on macOS, so it would seem to be outside the scope of Kinto, not that I don't want to implement it - but I don't really have a consensus sorta way to implement it since every Linux distro may use different defaults, as does Windows and if you were to really implement side by side windows the mac way then that would involve the mouse and holding down on the green button.

Not even sure how you would begin that in Linux lol.
https://apple.stackexchange.com/questions/50330/how-to-arrange-two-windows-easily-to-left-and-right-side. On the full screen and down sizing, again not really a macOS hotkey - more of a distro specific and preference thing. You are able to fork the project and edit the files you see fit. In this case ./setup.py and ./xkeysnail-config/kinto.py.

If you do come across some default hotkeys on a mac that is not included in Kinto please let me know in a new ticket or in this thread #44 so that I can include it in a future release.

**Comment by commadelimited** [comment]
Hey @rbreaves, to be clear I don't think this is the fault of Kinto at all not do I expect you to implement it for everyone. I make heavy use of an app called Divvy on my Mac that does a similar thing. Imagine my delight when it turns out Ubuntu has something similar built in?

Thanks for your suggestions!

**Comment by rbreaves** [comment]
Sure, I use a few 3rd party apps under macOS as well and used to make use similar functionality via BTT.

My recommendation if you want the Divvy like keybinds to be on the physical Left Ctrl key would be to bind Super+arrows in the kinto.py file as I mentioned earlier. This would also avoid any type of wordwise conflict functionality was well, because the newly mapped location for the actual Ctrl key does need those arrows for wordwise.

https://github.com/rbreaves/kinto/blob/master/xkeysnail-config/kinto.py

You'd make your GUI based Super remaps under this area, assuming you do not want to change the default binding of the particular DE you are on.

```
define_keymap(None,{
    # Basic App hotkey functions
    K("RC-Q"): K("Alt-F4"),
    K("RC-H"): K("Alt-F9"),
    # Divvy - Side by side windows
    K("Super-Left"): K("?? DE default ??"),
```

-----------------------------------------

And sorry I kept trying to skim your post on the Alt menu issue you were having and had to slow down and really read that. That is a problem that other users have reported and I would like to resolve it better. The wordwise situation is so different between macOS and linux it makes it difficult, and just about every Electron based app out there will make it difficult to resolve on almost anything but a case by case basis, if VS Code is anything to go by. There are 2 ways of resolving VS Code and the way I opted for is a method that works most the time, 90%+?, but will still activate the Alt menu at times.

It's not ideal, and possibly other apps than Electron based apps could be impacted, if you try to make use of wordwise. (Sublime Text 3 is not impacted at all.) I assume any actual Alt key combo that was native before will continue to work as expected whether it is pressed on the physical Alt key or a different modifier that it may be mapped to under Kinto.

The only other alternative I know of that could work would be a hybrid approach that would leverage the kinto xkb implementation for some select Alt based remaps, either globally or against specific apps. Although if you do it in an app specific way then some changes would need to happen to Kinto to allow for both implementations to run at the same time.

The XKB implementation of kinto, for whatever reason, manages to bypass the activation of the Alt based menu - but it cannot be combined with Kinto (xkeysnail/udev) unless the modifier keyswaps are removed from the xkb implementation.

TLDR; Try out the xkb implementation of Kinto. It is not as ideal and may lack some key remaps that Kinto xkeysnail has, but it won't activate your Alt menu's. At least I am pretty sure it won't.

**Comment by rbreaves** [comment]
@commadelimited A similar conversation about the Alt menu problem can be seen here #176.

**Comment by SaltwaterC** [comment]
TL;DR

In `~/.config/kinto/kinto.py` add under `define_keymap(re.compile(termStr, re.IGNORECASE),{`

```python
    K("LC-LEFT"): K("Super-LEFT"),
    K("LC-RIGHT"): K("Super-RIGHT"),
    K("LC-UP"): K("Super-UP"),
    K("LC-DOWN"): K("Super-DOWN")
```

Pro tip: don't forget the comma for the previous remap (like I did), then wondering why Kinto broke.

```bash
 sudo systemctl restart xkeysnail
```

**Comment by rbreaves** [comment]
For anyone whose terminal app expects Ctrl-Arrow key that’ll become broken, & I’m not aware of any specific app but the fear of that prevents me from merging this type of fix :/.

And since the creation of this ticket I do have better methods to support multiple DEs & different hotkeys fairly easily. Also xkb implementation of Kinto is no longer supported or offered, it’s all xkeysnail or ahk now.

**Comment by SaltwaterC** [comment]
Right. I'm a Gnome Terminal user, so for this particular use case, it works as expected, thought, YMMV.
